module/Application/src/Model/Service/ProcessingStatus/Service.php

```diff
@@ -63,49 +64,76 @@ public function setAwsSignatureV4(SignatureV4 $awsSignature)
     }
 
     /**
-     * @param $id
+     * @param $ids
      * @return mixed
      * @throws ApiProblemException
      * @throws Exception
      */
-    public function getStatus($id)
+    public function getStatuses($ids)
     {
-        if (is_numeric($id)) {
-            $id = 'A' . sprintf("%011d", $id);
-        }
-
-        $url = new Uri($this->processingStatusServiceUri . $id);
-
-        $request = new Request('GET', $url, $this->buildHeaders());
-
-        //---
+        // build request loop
+        $requests = [];
+        $successStatus = [];
 
         $provider = CredentialProvider::defaultProvider();
+        $credentials = $provider()->wait();
 
-        // Sign the request with an AWS Authorization header.
-        $signed_request = $this->awsSignature->signRequest($request, $provider()->wait());
-
-        //---
+        foreach ($ids as $id) {
 
-        $response = $this->httpClient->sendRequest($signed_request);
+            $prefixedId = $id;
 
-        $statusCode = $response->getStatusCode();
-
-        switch ($statusCode) {
-            case 200:
-                $status = $this->handleResponse($response);
+            if (is_numeric($id)) {
+                $prefixedId = 'A' . sprintf("%011d", $id);
+            }
 
-                $this->getLogger()->debug('Status ' . $status . ' returned from Sirius gateway for ID ' . $id);
-                return $status;
-            case 404:
-                // A 404 represents that details for the passed ID could not be found
-                $this->getLogger()->debug('No application status from Sirius gateway for ID ' . $id);
-                return null;
-            default:
-                $this->getLogger()
-                    ->err('Unexpected response from Sirius gateway: ' . (string)$response->getBody());
-                throw new ApiProblemException('Unexpected response from Sirius gateway: ' . $statusCode);
-        }
+            $url = new Uri($this->processingStatusServiceUri . $prefixedId);
+            $requests[$id] = new Request('GET', $url, $this->buildHeaders());
+            $requests[$id] = $this->awsSignature->signRequest($requests[$id], $credentials);
+        } //end of request loop
+
+        // build pool
+        $results = [];
+
+        $pool = new Pool($this->httpClient, $requests, [
+            'concurrency' => 10,
```

Comment:
```suggestion
            'concurrency' => 10,
            'options' => [
                'http_errors' => false,
            ],
```

When a 404 is returned, it looks to currently end up in `rejected`, as Guzzle's default behaviour is to treat a 404 as an error, rather than returning is for processing.

Reply:
Thanks Neil, did not realise